tensilelite/Tensile/Toolchain/Assembly.py

```diff
@@ -0,0 +1,226 @@
+                f"Error compressing code object via bundling: {err.output}\nFailed command: {' '.join(args)}"
+            )
+
+
+def _batchObjectFiles(objFiles: List[str], coPathDest: Union[Path, str], maxObjFiles: int=10000) -> List[str]:
+    numObjFiles = len(objFiles)
+    
+    if numObjFiles <= maxObjFiles:
+      return objFiles
+
+    batchedObjFiles = [objFiles[i:i+maxObjFiles] for i in range(0, numObjFiles, maxObjFiles)]
+    numBatches = int(math.ceil(numObjFiles / maxObjFiles))
+
+    newObjFiles = [str(coPathDest) + "." + str(i) for i in range(0, numBatches)]
+    newObjFilesOutput = []
+
+    for batch, filename in zip(batchedObjFiles, newObjFiles):
+      if len(batch) > 1:
+        args = [globalParameters["ROCmLdPath"], "-r"] + batch + [ "-o", filename]
+        print2(f"Linking object files into fewer object files: {' '.join(args)}")
+        subprocess.check_call(args)
+        newObjFilesOutput.append(filename)
+      else:
+        newObjFilesOutput.append(batchedObjFiles[0])
+
+    return newObjFilesOutput
+
+def buildAssemblyCodeObjectFiles(toolchain: AssemblyToolchain, kernels, writerAsm, outputPath, compress: bool=True):
+    
+    isAsm = lambda k: k["KernelLanguage"] == "Assembly"
+
+    extObj = ".o"
+    extCo = ".co"
+    extCoRaw = ".co.raw"
+
+    destDir = Path(ensurePath(os.path.join(outputPath, 'library')))
+    asmDir = Path(ensurePath(os.path.join(globalParameters["WorkingPath"], "assembly")))
+
+
+    archKernelMap = collections.defaultdict(list)
+    for k in filter(isAsm, kernels):
+      archKernelMap[tuple(k['ISA'])].append(k)
+
+    coFiles = []
+    for arch, archKernels in archKernelMap.items():
+      if len(archKernels) == 0:
+        continue
+
+      gfx = getGfxName(arch)
+
+      if globalParameters["MergeFiles"] or globalParameters["NumMergedFiles"] > 1 or globalParameters["LazyLibraryLoading"]:
+        objectFiles = [str(asmDir / (writerAsm.getKernelFileBase(k) + extObj)) for k in archKernels if 'codeObjectFile' not in k]
+
+        coFileMap = collections.defaultdict(list)
+
+        if len(objectFiles):
+          coFileMap[asmDir / ("TensileLibrary_"+ gfx + extCoRaw)] = objectFiles
+
+        for kernel in archKernels:
+          coName = kernel.get("codeObjectFile", None)
+          if coName:
+            coFileMap[asmDir / (coName + extCoRaw)].append(str(asmDir / (writerAsm.getKernelFileBase(kernel) + extObj)))
+
+        for coFileRaw, objFiles in coFileMap.items():
+
+          objFiles = _batchObjectFiles(objFiles, coFileRaw)
+          toolchain.link(objFiles, str(coFileRaw))
+
+          coFile = destDir / coFileRaw.name.replace(extCoRaw, extCo)
+          if compress:
+            toolchain.compress(str(coFileRaw), str(coFile), gfx)
+          else:
+            shutil.move(coFileRaw, coFile)
+
+          coFiles.append(coFile)
+      else:
+        # no mergefiles
+        def newCoFileName(kName):
+          if globalParameters["PackageLibrary"]:
```

Comment:
`PackageLibrary` removal is already in develop so this probably needs to be removed.